store/query/addRole.js

```diff
@@ -1,5 +1,23 @@
+const mercutio = require('mercutio');
+
 module.exports = async function (_id, role, scope) {
-	return await this.findOneAndUpdate({ _id }, {
+	const user = await this.findById(_id);
+	const oldRolesLength = user.roles.length;
+
+	await this.findOneAndUpdate({ _id }, {
 		$addToSet: { roles: { role, scope } },
 	});
+
+	const newUser = await this.findById(_id);
+
+	const newRoles = newUser.roles;
+	const m = mercutio(newRoles);
+	const rationalized = m.rationalize();
+	newUser.roles = rationalized.toArray();
+	newUser.save();
+
+	if (newUser.roles.length > oldRolesLength) {
+		return 201; // 201: created
```

Comment:
The store shouldn't return 201. This makes no sense. The store does not know that it is being used for HTTP calls. 

However, it doesn't make sense either, that it returns a boolean whether it added something or not. It should just return the new role array.